### PR TITLE
Add "Bearer" as an auth option

### PR DESCRIPTION
**Pull request by hamzahejja**
- This PR addresses issue #1919 
- Adds Bearer Token Authorization to Client Requests.
- Added needed test.

### CODE REVIEW

**Comment by Nyholm** [comment]
I don’t see how this is a more convenient way than just setting the authorization header yourself. If we feel that we really want to include this, then I’m fine with the implementation. (Good job btw)

I’ve added the milestone 7.1.

**Comment by hamzahejja** [comment]
@Nyholm I agree with what you've said, it makes sense and is sufficient to only set the `Authorization` header with the bearer token. The reason why I implemented this was per issue #1919 which was **Accepting PRs**. I Appreciate the fast feedback!

**Comment by twoixter** [comment]
@Nyholm FWIW, I found this PR while wondering why `RequestOptions::AUTH` didn't recognise `bearer` as built-in, as it is widely used because of OAuth 2.0 nowadays. I mean, it's not just a user-defined method in which you're expected to manually add the `Authorization` header.

It is a great feature for tools that saves configuration options so that I don't have to `switch (...)` or similar just for the sake of adding the header or using `RequestOptions::AUTH`.

File: src/Client.php [hunk 1]
```diff
@@ -405,6 +405,11 @@ private function applyOptions(RequestInterface $request, array &$options): Reque
                     $options['curl'][CURLOPT_HTTPAUTH] = CURLAUTH_NTLM;
                     $options['curl'][CURLOPT_USERPWD] = "$value[0]:$value[1]";
                     break;
+                case 'bearer':
+                    // Ensure that we don't have the Authorization header in
+                    // different case and set the new value wih the token
```
Comment (davidgrayston) [inline src/Client.php:410]:
```suggestion
                    // different case and set the new value with the token
```
Noticed this minor typo

**Comment by Nyholm** [comment]
Sorry for the long delay. Could you please rebase this PR?

**Comment by gmponos** [comment]
Hi.. @hamzahejja thank you for the contribution.. 

This was my bad marking the original issue as `accept PRs`

For now this seems like a not needed change. As @Nyholm said developers can easily set the `Authorization` header

**Pull request closed without merging**